clients/venice-samza/src/main/java/com/linkedin/venice/samza/VeniceSystemProducer.java

```diff
@@ -744,6 +770,38 @@ private static <T> byte[] serializePrimitive(T input, DatumWriter<T> writer) {
     return out.toByteArray();
   }
 
+  private Object convertWriteComputeRecordToFullPut(
+      int derivedSchemaId,
+      int baseSchemaId,
+      Object incomingWriteValueObject) {
+    Schema baseSchema = valueSchemaIds.entrySet()
```

Comment:
nit: maybe we can build a reverse index from id -> value schema. so not to filter on every operation.

Reply:
Yeah might as well.  Will add.